jena-geosparql/src/main/java/org/apache/jena/geosparql/geo/topological/SpatialObjectGeometryLiteral.java

```diff
@@ -132,8 +133,22 @@ protected static final SpatialObjectGeometryLiteral retrieve(Graph graph, Node t
         } else {
             //Target is not a Feature or Geometry but could have Geo Predicates.
             if (graph.contains(targetSpatialObject, SpatialExtension.GEO_LAT_NODE, null) && graph.contains(targetSpatialObject, SpatialExtension.GEO_LON_NODE, null)) {
-                Node lat = graph.find(targetSpatialObject, SpatialExtension.GEO_LAT_NODE, null).next().getObject();
-                Node lon = graph.find(targetSpatialObject, SpatialExtension.GEO_LON_NODE, null).next().getObject();
+
+                //Extract Lat coordinate.
+                ExtendedIterator<Triple> latIter = graph.find(targetSpatialObject, SpatialExtension.GEO_LAT_NODE, null);
+                Node lat = latIter.next().getObject();
+
+                //Extract Lon coordinate.
+                ExtendedIterator<Triple> lonIter = graph.find(targetSpatialObject, SpatialExtension.GEO_LON_NODE, null);
+                Node lon = lonIter.next().getObject();
+
+                //Ensure that only a single Latitude and Longitude are present for the subject.
+                if (latIter.hasNext() || lonIter.hasNext()) {
+                    latIter.close();
+                    lonIter.close();
+                    throw new DatatypeFormatException(targetSpatialObject.getURI() + " has more than one geo:lat or geo:lon property.");
+                }
+
```

Comment:
Sorry to be late to the game. 

A comment: a useful pattern is 

    ExtendedIterator<> iter = ...;
    try { 
       ... process ...
    } finally { iter.close(); }

There is no harm calling close multiple times or on already closed iterators.